projects/plutus/src/ltc_adc.c

```diff
@@ -14,24 +14,73 @@ uint8_t s_filter_modes[NUM_LTC_ADC_FILTER_MODES] = {
   LTC2484_REJECTION_60HZ,
 };
 
-static GPIOSettings s_settings = { .direction = GPIO_DIR_IN,
-                                   .state = GPIO_STATE_HIGH,
-                                   .resistor = GPIO_RES_NONE,
-                                   .alt_function = GPIO_ALTFN_0 };
+typedef struct {
+  StatusCode status;
+  int32_t value;
+} LtcAdcBuffer;
+
+static LtcAdcSettings *s_config;
```

Comment:
Can't you just move these into the config struct? Rename it to storage or something and use a initial setting struct to populate the storage.